### DISCUSSION TITLE
Formatters

### DISCUSSION THREAD

**Discussion by YSCohen** [hylang/hy#2253] [General]
Are there any good Hy formatters? Preferably this would be a Python library so I can just `pip install`. Thanks!

**Comment by allison-casey** [comment]
no there are no dedicated Hy formatters at the moment. your best bet is using something like emac's lispy indent or look into some more generic lisp formatters

**Comment by YSCohen** [comment]
I use VS Code, and this is my first Lisp. Any suggested Lisp formatters?

**Comment by allison-casey** [comment]
I can't really speak to vs code, but you might be able to piggyback off of an existing common lisp extension's formatter. There's also [scmindent](https://github.com/ds26gte/scmindent) seems to work on hy well enough from my cursory testing, even without customizing the lispwords. I imagine it's easy enough to get vscode to pipe the buffer to scmindent, but again, i'm not a VSCode person

**Comment by Kodiologist** [comment]
Historically, Hy borrowed a lot of syntax from Clojure. This is less true now (if only because we now lean towards imitating Python more transparently), but if you have to choose from tools for an existing specific Lisp, Clojure probably remains your best bet.

I think [Hydiomatic](https://github.com/hylang/hydiomatic) may have had some autoformatting capabilities, but it's a few years out of date, so there would be work to do to get it to work with the latest Hy.
